mailpoet/lib/EmailEditor/Engine/Renderer/BlockRenderer.php

```diff
@@ -2,6 +2,28 @@
 
 namespace MailPoet\EmailEditor\Engine\Renderer;
 
-interface BlockRenderer {
-  public function render(string $blockContent, array $parsedBlock): string;
+use MailPoet\DI\ContainerWrapper;
+use MailPoet\EmailEditor\Engine\SettingsController;
+
+abstract class BlockRenderer {
+  /** @var SettingsController */
+  protected $settingsController;
+
+  public function __construct() {
+    $this->settingsController = ContainerWrapper::getInstance()->get(SettingsController::class);
+  }
+
+  public abstract function render(string $blockContent, array $parsedBlock): string;
+
+  public function getLayoutWidth(): string {
+    $layoutStyles = $this->settingsController->getEmailLayoutStyles();
+    $width = $this->parseNumberFromStringWithPixels($layoutStyles['width']);
+    $width -= $this->parseNumberFromStringWithPixels($layoutStyles['padding']['left']);
+    $width -= $this->parseNumberFromStringWithPixels($layoutStyles['padding']['right']);
+    return "{$width}px";
+  }
+
+  private function parseNumberFromStringWithPixels(string $string): float {
+    return (float)str_replace('px', '', $string);
+  }
```

Comment:
This works only with data from the settings controller so I guess we could move it into the controller.